bokehjs/src/coffee/api/typings/models/tools.d.ts

```diff
@@ -78,19 +78,23 @@ declare namespace Bokeh {
     export var BoxSelectTool: { new(attributes?: IBoxSelectTool, options?: ModelOpts): BoxSelectTool };
     export interface BoxSelectTool extends SelectTool, IBoxSelectTool {}
     export interface IBoxSelectTool extends ISelectTool {
+        callback?: Callback | ((source: this) => void);
```

Comment:
There's two issues here (thanks for bringing this up @mattpap )

1. In terms of unifying the select tools: the TapTool currently returns the DataSource as the cb_obj in the callback (which is required by the working of OpenURL), while all of the other tools return the tool instance. Should all of the tools return the data source or the tool instance (which is perhaps less helpful, but more consistent with the Bokeh pattern of returning the instance of the object that kicked off the callback).

2. In terms of where to place the `callback` attr: the class hierarchy of the tools in python and JS don't match up very well, so there's no python `SelectTool` model.